### PR TITLE
Refactor/dsg generate server

### PR DESCRIPTION
**Pull request by abrl91**
Close: https://github.com/amplication/amplication/issues/7405

### Summary
🆕🛠️🧪

Refactored the data service generator to use DTOs from the context object and to support generating server and admin modules based on settings. Added logging and imports to reflect the changes. Added test cases for creating data services without server or admin UI.

> _Sing, O Muse, of the skillful coder who devised_
> _A way to create data services with various options_
> _And tested them with snapshots, as true as the sun's rays_
> _That shine upon the earth from the golden chariot of Helios._

### Walkthrough
*  Refactor `createDataService` function to create DTOs from entities and store them in `context` object, and to check `generateServer` and `generateAdminUI` settings from `appInfo` object before creating server and admin modules ([link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-49351fe06b663c0ae76597a6a63b848c9c27030188cf288f20d58a2ed7120f28R11-R12), [link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-49351fe06b663c0ae76597a6a63b848c9c27030188cf288f20d58a2ed7120f28L50-R74))
* Remove redundant creation of DTOs from `createAdminModulesInternal` and `createServerInternal` functions, and use `context.DTOs` property instead ([link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-7e369d3fdc3528cd25d13aadc06d5132caf0697deaf539d51f7bbc36f53099d6L42-R42), [link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-7e369d3fdc3528cd25d13aadc06d5132caf0697deaf539d51f7bbc36f53099d6L81-R77), [link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaL58-R59))
* Remove unused `createDTOs` import from `create-server.ts` file ([link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaL9-R9))
* Add `generateServer` property to `appInfo` object in `appInfo.ts` file for testing purposes ([link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-d1e9a64d52d16f55650a0109dbf9b18b466d018a454a5c7e3ddfd47b888a4baeR27))
* Add test case for creating data service without generating admin UI in `create-data-service-without-admin-ui.spec.ts` file, and use snapshot to compare expected and actual output ([link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-5e15d6846d182a844baaa61d3513546b75769f11e34a6e40e2d3fe546d3d9e4fR1-R54))
* Add test case for creating data service without generating server in `create-data-service-without-server.spec.ts` file, and use snapshot to compare expected and actual output ([link](https://github.com/amplication/amplication/pull/7428/files?diff=unified&w=0#diff-dbfe6e1e3395178c93025e38364c179a6075cb800c8ce91e1a6da428ae7ee98aR1-R54))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/data-service-generator/src/admin/create-admin.ts [hunk 1]
```diff
@@ -39,13 +39,7 @@ export function createAdminModules(): Promise<ModuleMap> {
...
 async function createAdminModulesInternal(): Promise<ModuleMap> {
   const context = DsgContext.getInstance;
-  const {
-    entities,
-    roles,
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    DTOs,
-    clientDirectories,
-  } = context;
+  const { entities, roles, clientDirectories } = context;
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/data-service-generator/src/admin/create-admin.ts:42]: but I wanted to remove the ignore :)

[the remark this answers is not in the document]
Comment (abrl91) [reply packages/data-service-generator/src/admin/create-admin.ts:42]: I agree if the name of the variable is dtos but its DTOs

[the remark this answers is not in the document]
Comment (overbit) [reply packages/data-service-generator/src/admin/create-admin.ts:42]: Yeah, my suggestion was to remove the const context too 😁

[the remark this answers is not in the document]
Comment (overbit) [reply packages/data-service-generator/src/admin/create-admin.ts:42]: We can add a rule to allow DTOs

[the remark this answers is not in the document]
Comment (abrl91) [reply packages/data-service-generator/src/admin/create-admin.ts:42]: let's open an issue

File: packages/data-service-generator/src/tests/__snapshots__/create-data-service-with-auth-jwt.spec.ts.snap
Comment (mulygottlieb) [inline packages/data-service-generator/src/tests/__snapshots__/create-data-service-with-auth-jwt.spec.ts.snap:1]: I don't understand the changes that happened to this snapshot - why were those fields removed?

**Review by overbit — APPROVED** [review]

**Comment by overbit** [comment]
@abrl91 remember to open the issue to clean the `spread` and `const`

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**